File: src/main/java/training/katas/BitCounting/bit_counting_test.go
package BitCounting

import (
	// . "codewarrior/kata"
	"github.com/dailymotion/allure-go"
	"github.com/joho/godotenv"
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
	"log"
	"os"
	"testing"
)

var _ = Describe("CountBits()", func() {
	It("basic tests", func() {
		Expect(CountBits(0)).To(Equal(0))
		Expect(CountBits(4)).To(Equal(1))
		Expect(CountBits(7)).To(Equal(3))
		Expect(CountBits(9)).To(Equal(2))
		Expect(CountBits(10)).To(Equal(2))
	})
})

const (
	Epic    = "epic Training"
	Feature = "feature Katas"
	Story   = "story Bit Counting"
)

// https://towardsdatascience.com/use-environment-variable-in-your-next-golang-project-39e17c3aaa66
// use godot package to load/read the .env file and
// return the value of the key
func goDotEnvVariable(key string) string {

	// load .env file
	err := godotenv.Load("../../../../../../variables.env")

	if err != nil {
		log.Fatal(err)
		log.Fatalf("Error loading .env file")
	}

	return os.Getenv(key)
}

func dotest(input uint, exp int) {
	var ans = CountBits(input)
	Expect(ans).To(Equal(exp))
}

func TestStepBitCounting(t *testing.T) {

	dotenv := goDotEnvVariable("ALLURE_RESULTS_PATH")
	allure.Test(t,
		allure.Epic(Epic),
		allure.Feature(Feature),
		// // allure.Story(Story),
		// allure.Description("test Maximum subarray 01"),
		allure.Action(func() {
			Expect(CountBits(0)).To(Equal(0))
			Expect(CountBits(4)).To(Equal(1))
			Expect(CountBits(7)).To(Equal(3))
			Expect(CountBits(9)).To(Equal(2))
			Expect(CountBits(10)).To(Equal(2))
		}))

}

File: src/main/java/training/katas/BitCounting/bit_counting.test.js
/*
const { assert } = require("chai")

describe("Basic tests", () => {
        it("Testing for fixed tests", () => {
            assert.strictEqual(countBits(0), 0);
            assert.strictEqual(countBits(4), 1);
            assert.strictEqual(countBits(7), 3);
            assert.strictEqual(countBits(9), 2);
            assert.strictEqual(countBits(10), 2);
        })
    })
*/

// const chai = require("chai");
const countBits = require("./bit_counting").countBits;
const { assert, expect } = require("chai")
    // chai.config.truncateThreshold = 0;
    // const expect = require("chai").expect;

describe("Bit Counting tests", function() {
    it("Tests", function() {
        assert.strictEqual(countBits(0), 0);
        assert.strictEqual(countBits(4), 1);
        assert.strictEqual(countBits(7), 3);
        assert.strictEqual(countBits(9), 2);
        assert.strictEqual(countBits(10), 2);
    });
});
test("0 -> 0", () => {
    expect(countBits(0)).to.equal(0);
});
test("4 -> 1", () => {
    expect(countBits(4)).to.equal(1);
});
test("7 -> 3", () => {
    expect(countBits(7)).to.equal(3);
});
test("9 -> 2", () => {
    expect(countBits(9)).to.equal(2);
});
test("10 -> 2", () => {
    expect(countBits(10)).to.equal(2);
});
test("4 -> 1", () => {
    expect(countBits(4)).to.equal(1);
});